### PR TITLE
Vector / Matrix updates and fixes

### PR DESCRIPTION
**Pull request by bernardkwok**
- Add in vec2<->vec3 and vec3<->vec4 "convert" support
- Fix up to have matrix33 types emit proper code for OSL. GLSL does not require this.
-- Pad matrix33 to be a matrix(44) in custom emit code.
- Fix up GLSL matrix vector ordering so it does the proper math for points and vector vs normals. (M*V)
vs (V*M), where V*M is uses transpose matrix logic.
- Add in support for transformvector(vec2, matrix33) and transformvector(vec3,matrix44)
- Update tests to use new convert and transformvector variants.

### CODE REVIEW

File: documents/Libraries/stdlib/genosl/mx_transformvector2_matrix33.osl [hunk 1]
```diff
@@ -0,0 +1,7 @@
+void mx_transformvector2_matrix33(vector2 val, matrix m, output vector2 result)
+{
+    vector _vec3 = vector(val.x, val.y, 0.0);
```
Comment (feldstj) [inline documents/Libraries/stdlib/genosl/mx_transformvector2_matrix33.osl]: Why are the glsl and osl variants of this slightly different? Namely the vec3 being created ends in a 0 vs a 1. Also not to nitpick, but you called it _val in the glsl variant and _vec3 here.

Comment (bernardkwok) [reply documents/Libraries/stdlib/genosl/mx_transformvector2_matrix33.osl]: Oh yes. This is incorrect. I forgot to fix it and make things consistent as well.

File: documents/TestSuite/stdlib/math/transform.mtlx [hunk 2]
```diff
@@ -8,113 +8,196 @@ Basic transform function tests each test is in a separate graph for each variati
...
     <output name="out" type="vector4" nodename="transformpoint1" />
   </nodegraph>
-  <!--
-  <nodegraph name="transform_point_vector3M">
+
+  <nodegraph name="tf_point_vector3M">
     <transformpoint name="transformpoint1" type="vector3">
-	    <input name="in" type="vector3" value="1.0, 0.5, 0.0"/>
-	    <input name="mat" type="matrix33" value="0.5, 0.0, 0.0,  0.0, 2.0, 0.0,  0.0, 0.0, 0.0"/>
+	    <input name="in" type="vector3" value="1.0, 0.5, 0.0" nodename="position1"/>
```
Comment (feldstj) [inline documents/TestSuite/stdlib/math/transform.mtlx]: minor: you seemed to fix the indentation on other ones, but not this one.

File: documents/TestSuite/stdlib/math/transform.mtlx [hunk 3]
```diff
@@ -8,113 +8,196 @@ Basic transform function tests each test is in a separate graph for each variati
...
+    </convert>
   </nodegraph>
-  <!--
-  <nodegraph name="transform_vector_vector3">
+
+  <!-- vector transform -->
+  <nodegraph name="tf_vector_vector3">
     <transformvector name="transformvector1" type="vector3">
-	    <input name="in" type="vector3" value="1.0, 0.5, 0.0"/>
+	    <input name="in" type="vector3" value="1.0, 0.5, 0.0" nodename="normal1"/>
```
Comment (feldstj) [inline documents/TestSuite/stdlib/math/transform.mtlx]: minor: indentation

File: documents/TestSuite/stdlib/math/transform.mtlx [hunk 4]
```diff
@@ -8,113 +8,196 @@ Basic transform function tests each test is in a separate graph for each variati
...
     <output name="out" type="vector4" nodename="transformnormal1" />
   </nodegraph>
-  <!--
-  <nodegraph name="transform_normal_vector3M">
+
+  <nodegraph name="tf_normal_vector3M">
     <transformnormal name="transformnormal1" type="vector3">
-	    <input name="in" type="vector3" value="1.0, 0.5, 0.0"/>
-	    <input name="mat" type="matrix33" value="0.5, 0.0, 0.0,  0.0, 2.0, 0.0,  0.0, 0.0, 0.0"/>
+	    <input name="in" type="vector3" value="1.0, 0.5, 0.0" nodename="normal1"/>
```
Comment (feldstj) [inline documents/TestSuite/stdlib/math/transform.mtlx]: minor: indentation

**Pull request merged**